docs/manual-test-plan.md

```diff
@@ -0,0 +1,809 @@
+  --header 'Content-Type: application/json' \
+  --data '{
+    "action": "restore",
+    "enrollment_code": "PY4ngLKw"
+}' \
+  --header "Authorization: Bearer $BEARER_TOKEN"
+```
+
+Verify the response has status 200 and the following json body:
+
+```json
+{
+  "success": true
+}
+```
+
+Visit the web UI as the "apicreated4" user again,
+and verify the user is no longer enrolled in the demo course.
+
+Enroll the user in the demo course again using the same (restored) code:
+
+```sh
+curl -X POST 'http://local.openedx.io:8000/appsembler_api/v0/enrollment-codes/enroll-user' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "email": "apicreated1@example.com",
+    "enrollment_code": "PY4ngLKw"
+}' \
+  --header "Authorization: Bearer $BEARER_TOKEN"
+```
+
+Verify a success response is returned:
+
+```json
+{
+  "success": true
+}
+```
+
+In the web UI, verify the user is enrolled again in the demo course.
+
+Now cancel the enrollment code:
+
+```sh
+curl -X POST 'http://local.openedx.io:8000/appsembler_api/v0/enrollment-codes/status' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "action": "cancel",
+    "enrollment_code": "PY4ngLKw"
+}' \
+  --header "Authorization: Bearer $BEARER_TOKEN"
+```
+
+Verify the response has status 200 and the following json body:
+
+```json
+{
+  "success": true
+}
+```
+
+In the web UI, verify the user is once again no longer enrolled in the demo course.
+
+One final time, attempt to enroll the user in the course using the same code:
+
+```sh
+curl -X POST 'http://local.openedx.io:8000/appsembler_api/v0/enrollment-codes/status' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "action": "restore",
+    "enrollment_code": "PY4ngLKw"
+}' \
+  --header "Authorization: Bearer $BEARER_TOKEN"
+```
+
+Verify the response has status 400 and the following json body:
+
+```json
+{
+  "success": false,
+  "reason": "Enrollment code not found"
+}
+```
+
+### Test the accounts analytics endpoint
+
+This shows information about user accounts, and can filter by date joined.
+
+You can get a list of all the users on the platform with:
+
+```sh
+curl -X GET 'http://local.openedx.io:8000/appsembler_api/v0/analytics/accounts/batch' \
+  --header 'Content-Type: application/json' \
+  --data '' \
+  --header "Authorization: Bearer $BEARER_TOKEN"
+```
+
+Verify the response has status 200 and the body has a json list of all users:
+
+```json
+[
+  {
+    "id": 2,
+    "username": "login_service_user",
+    "email": "login_service_user@fake.email",
+    "is_active": true,
+    "date_joined": "2025-09-29T06:35:08.751215Z"
+  },
+  {
+    "id": 3,
+    "username": "cms",
+    "email": "cms@openedx",
+    "is_active": true,
+    "date_joined": "2025-09-29T06:36:27.318723Z"
+  },
+  {
+    "id": 5,
+    "username": "apicreated4",
+    "email": "apicreated1@example.com",
+    "is_active": true,
+    "date_joined": "2025-09-29T07:05:03.092197Z"
+  },
+  ...
```

Comment:
```suggestion
  // ...
]
```

Reply:
The renderer on GitHub does not like the ellipsis here:

<img width="697" height="253" alt="image" src="https://github.com/user-attachments/assets/33c7319e-84ef-4e60-8900-be2008cfac2f" />

And on the diff from the review page:

<img width="697" height="253" alt="image" src="https://github.com/user-attachments/assets/5d4caa1e-aa15-415b-9b56-1fcf69f7ebdb" />

Reply:
Ah that really messes up the diff highlighting, wow. I was hoping for the simple red ellipsis to show it's been truncated. I like your json5 method, thanks. :+1: :+1: